tools/tck/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala

```diff
@@ -140,39 +144,50 @@ object CypherTCK {
 
       val scenarioSteps: List[Step] = step.getText match {
         // Given
-        case emptyGraphR()     => List.empty
+        case emptyGraphR() => List.empty
         case namedGraphR(name) => List(Execute(NamedGraphs.graphs(name), InitQuery))
-        case anyGraphR()       => List(Execute(NamedGraphs.graphs.values.head, InitQuery))
+        case anyGraphR() => List(Execute(NamedGraphs.graphs.values.head, InitQuery))
 
         // And
-        case initQueryR()                   => List(Execute(queryFromStep, InitQuery))
-        case parametersR()                  => List(Parameters(parseParameters))
+        case initQueryR() => List(Execute(queryFromStep, InitQuery))
+        case parametersR() => List(Parameters(parseParameters))
         case installedProcedureR(signature) => List(RegisterProcedure(signature, parseTable()))
 
         // When
-        case executingQueryR()        => List(Measure, Execute(queryFromStep, ExecQuery))
+        case executingQueryR() => List(Measure, Execute(queryFromStep, ExecQuery))
         case executingControlQueryR() => List(Execute(queryFromStep, ExecQuery))
 
         // Then
-        case expectEmptyResultR()          => List(ExpectResult(CypherValueRecords.empty))
-        case expectResultR()               => List(ExpectResult(parseTable()))
-        case expectSortedResultR()         => List(ExpectResult(parseTable(), sorted = true))
+        case expectEmptyResultR() => List(ExpectResult(CypherValueRecords.empty))
+        case expectResultR() => List(ExpectResult(parseTable()))
+        case expectSortedResultR() => List(ExpectResult(parseTable(), sorted = true))
         case expectResultUnorderedListsR() => List(ExpectResult(parseTable(orderedLists = false)))
         case expectErrorR(errorType, time, detail) =>
-          List(ExpectError(errorType, time, detail).validate(), SideEffects().fillInZeros)
+          val expectedError = ExpectError(errorType, time, detail)
+          if (shouldValidate) {
+            expectedError.validate match {
+              case None => // No problem
+              case Some(errorMessage) =>
+                throw new InvalidFeatureFormatException(
+                  s"""Invalid error format in scenario "${pickle.getName}" from feature "$featureName":
+                    $errorMessage
+                    If this is a custom error, then disable this validation with tag "@allowCustomErrors"""")
```

Comment:
How are the custom errors validated?

Reply:
The flag disables error format validation.